src/main/java/org/commcare/xml/DetailParser.java

```diff
@@ -58,6 +61,16 @@ public Detail parse() throws InvalidStructureException, IOException, XmlPullPars
         String focusFunction = null;
 
         while (nextTagInBlock("detail")) {
+            if (GlobalParser.NAME_GLOBAL.equals(parser.getName().toLowerCase())) {
+                try {
+                    checkNode(GlobalParser.NAME_GLOBAL);
+                    global = new GlobalParser(parser).parse();
+                    parser.nextTag();
+
+                } catch (InvalidStructureException e) {
+                    System.out.println("Lookup block not found " + e);
```

Comment:
where does this error come from? This catch should almost certainly kill the parse and result in a global InvalidStructureException, so it would be better to not be caught at all

Reply:
yeah makes sense, I kind of copied it from the parsing below for `lookup`. Another thing that doesn't make sense to me is we ae doing a `checkNode(GlobalParser.NAME_GLOBAL);` just after `if (GlobalParser.NAME_GLOBAL.equals(parser.getName().toLowerCase())) {` . Do you know why we are following that pattern ?